### PR TITLE
#248: Standardize error messages across codebase

### PR DESCRIPTION
**Pull request by lucifercr07**

### CODE REVIEW

**Comment by lucifercr07** [comment]
@soumya-codes would like to discuss once regarding the changes for Standardization of errors [issue](https://github.com/DiceDB/dice/issues/248)
Wanted to check on few points:
1. Should we follow a `enum` like structure to standardise use of Command names, info etc. across the codebase?
2. Redis follows an approach where it passes the `client` pointer across which holds all details of the command being executed? Should we also try to follow the same? As currently under each method we need to hardcode the command name when throwing error.

Please review once haven't changed all error messages as wanted to check if these changes look good or we need to create some other framework.

File: core/diceerrors/diceerrors.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func newDiceErr(message string) *DiceError {
+	return &DiceError{
+		message:        message,
+		encodedMessage: []byte(fmt.Sprintf("-%s\r\n", message)),
+	}
+}
+
+func NewErrArity(cmdName string) []byte {
```
Comment (soumya-codes) [inline core/diceerrors/diceerrors.go:43]: unless we are using these methods outside of the package, let us have them scoped to the package. We need to enforce this across the project @JyotinderSingh

**Review by soumya-codes — COMMENTED** [review]
@lucifercr07 i agree with the overall approach. please go ahead with the same. have left a few comments.

**Comment by lucifercr07** [comment]
@soumya-codes @JyotinderSingh please review once

File: core/bloom.go [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 import (
-	"errors"
 	"fmt"
 	"hash"
 	"math"
 	"math/rand"
 	"strconv"
 
+	DiceErrors "github.com/dicedb/dice/core/diceerrors"
```
Comment (soumya-codes) [inline core/bloom.go:10]: nit: the package alias seems redundant

File: core/eval.go [hunk 3]
```diff
@@ -11,6 +10,8 @@ import (
 	"syscall"
 	"time"
 
+	DiceErrors "github.com/dicedb/dice/core/diceerrors"
```
Comment (soumya-codes) [inline core/eval.go:13]: nit: the package alias seems redundant

**Review by soumya-codes — COMMENTED** [review]
The changes look good. Have left couple of minor messages. We have newly introduced local linter integration. Please make sure the changes are compatible with the linter.

**Comment by AshwinKul28** [comment]
@lucifercr07 Please check the linter warnings. You can easily fix them. Thanks

**Review by soumya-codes — APPROVED** [review]
LGTM

File: core/diceerrors/diceerrors.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+func NewErrWithFormattedMessage(errMsgFmt string, args ...interface{}) []byte {
+	if len(args) > 0 {
+		errMsgFmt = fmt.Sprintf(errMsgFmt, args...)
+	}
+
+	return NewErrWithMessage(errMsgFmt)
+}
+
+func NewErrArity(cmdName string) []byte {
```
Comment (yashs360) [inline core/diceerrors/diceerrors.go:60]: Whats `Arity`?

**Review by yashs360 — APPROVED** [review]
Looks good. Thanks @lucifercr07

**Comment by JyotinderSingh** [comment]
Looks like the changes went out of sync, could you please rebase and check for any new errors that may have been added? I'll try to merge the PR soon after that so we don't have to keep rebasing.

**Comment by JyotinderSingh** [comment]
There seems to be a test failure, could you please check?

**Comment by lucifercr07** [comment]
@JyotinderSingh Have pushed the fix on my fork branch, was this PR accidentally closed? Shall I raise a new PR?

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh Have pushed the fix on my fork branch, was this PR accidentally closed? Shall I raise a new PR?

Might have closed it by mistake, sorry!

**Comment by lucifercr07** [comment]
@JyotinderSingh build passed, please merge.

**Comment by JyotinderSingh** [comment]
Thanks a ton for this massive effort!

**Comment by JyotinderSingh** [comment]
#248

**Pull request merged**